lib/internal/streams/writable.js

```diff
@@ -565,12 +567,12 @@ function clearBuffer(stream, state) {
 
     resetBuffer(state);
   } else {
-    do {
+    for (; i < buffered.length && !state.writing; i++) {
```

Comment:
Is a do while slower than a for loop? Doesnt make much sense tbh.. 
What about first checking if !state.writing and then i < buffered.length?

Reply:
The performance gain is due to the position of the `state.bufferProcessing = true;`, not due to this change. It would be nice to have a micro-benchmark to validate if `do while` performance against a `for loop`. My gut tells me it would be slower, but not sure without a benchmark.

Reply:
Imho while loops are the fastest if the amount of iterations is unknown and could change while looping.

Reply:
> The performance gain is due to the position of the state.bufferProcessing = true;, not due to this change.

If that's the case, and just for clarity, could that change be removed?

Reply:
I removed the benchmarks due to the flakiness of the results, and updated the description.